text/0000-stabilize-arbitrary-self-types.md

```diff
@@ -283,79 +264,55 @@ impl Foo {
-would prevent Rust from ever adding more methods to the raw pointer primitive
-type. That doesn't feel like it will be OK, and so there is no known path
-to stabilizing raw pointer self types. Therefore, we choose to just remove
-this facility.
+However, as discussed under [method Shadowing section](#method-shadowing) this would prevent Rust from ever adding more methods to the raw pointer primitive type. That doesn't feel like it will be OK, and so there is no known path to stabilizing raw pointer self types. Therefore, we choose to just remove this facility.
 
 ## Allow implementation of `Receiver` for foreign types
 
-An alternative workaround for the removal of this "raw pointer self type"
-facility is to allow explicit implementation of `Receiver` for raw pointers.
+An alternative workaround for the removal of this "raw pointer self type" facility is to allow explicit implementation of `Receiver` for _specific_ types of raw pointer.
 
 ```rust
 impl Receiver for *const MyType {
     type Target = MyType;
 }
 ```
 
-This currently falls foul of the orphan rule. We could add an exception
-just as we have for [traits parameterized by local types](https://blog.rust-lang.org/2020/01/30/Rust-1.41.0.html#relaxed-restrictions-when-implementing-traits)
-but this would be complex in itself.
+This currently falls foul of the orphan rule. We could add an exception just as we have for [traits parameterized by local types](https://blog.rust-lang.org/2020/01/30/Rust-1.41.0.html#relaxed-restrictions-when-implementing-traits) but this would be complex in itself.
 
 ## Implement for `Weak` and `NonNull`
 
-`Weak<T>` and `NonNull<T>` were not supported by the prior unstable arbitrary self tpes
-support, but they share the property that it may be desirable to implement
-method calls to `T` using them as self types. Unfortunately they also share the property that these types
-have many Rust methods using `self`, `&self` or `&mut self`. If we added to the set of Rust methods in future,
-we'd [shadow any such method calls](#method-shadowing). We can't implement `Receiver` for these types unless
-we come up with a policy that all subsequent additions to these types would
-instead be associated functions.
+`Weak<T>` and `NonNull<T>` were not supported by the prior unstable arbitrary self tpes support, but they share the property that it may be desirable to implement method calls to `T` using them as self types. Unfortunately they also share the property that these types have many Rust methods using `self`, `&self` or `&mut self`. If we added to the set of Rust methods in future, we'd [shadow any such method calls](#method-shadowing). We can't implement `Receiver` for these types unless we come up with a policy that all subsequent additions to these types would instead be associated functions. That would make the future APIs for these types a confusing mismash of methods and associated functions, and the extra user complexity doesn't seem merited.
```

Comment:
```suggestion
`Weak<T>` and `NonNull<T>` were not supported by the prior unstable arbitrary self types support, but they share the property that it may be desirable to implement method calls to `T` using them as self types. Unfortunately they also share the property that these types have many Rust methods using `self`, `&self` or `&mut self`. If we added to the set of Rust methods in future, we'd [shadow any such method calls](#method-shadowing). We can't implement `Receiver` for these types unless we come up with a policy that all subsequent additions to these types would instead be associated functions. That would make the future APIs for these types a confusing mismash of methods and associated functions, and the extra user complexity doesn't seem merited.
```